src/workerd/api/pyodide/pyodide.c++

```diff
@@ -117,6 +119,18 @@ int PyodideMetadataReader::readMemorySnapshot(int offset, kj::Array<kj::byte> bu
   return readToTarget(KJ_REQUIRE_NONNULL(memorySnapshot), offset, buf);
 }
 
+kj::Array<kj::String> PyodideMetadataReader::getTransitiveRequirements() {
+  auto packages = parseLockFile(packagesLock);
+  auto depMap = getDepMapFromPackagesLock(*packages);
+
+  auto allRequirements = getPythonPackageNames(*packages, depMap, requirements, packagesVersion);
```

Comment:
I really wish we could instead add to jsg the option to return HashSet as JsSet. we are converting a set to an array only to convert it back to a set.
Is it really that difficult to add native jsg support for sets?

Reply:
I haven't tried, but indeed it does sound like it should be fairly simple. I'll have a look.

Reply:
This should do it https://github.com/cloudflare/workerd/pull/3550